File: backend/order-service/src/main/java/com/abdelkhalek/storehub/order/cart/dtos/CartResponse.java
package com.abdelkhalek.storehub.order.cart.dtos;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record CartResponse(
        UUID cartId,
        List<CartItemResponse> items,
        BigDecimal originalTotal,
        BigDecimal finalTotal,
        BigDecimal totalDiscount,
        UUID storeId

) {
}

File: frontend/src/app/cart/model/cart-response.ts
import { UUID } from 'node:crypto';

export interface CartResponse {
  cartId: string;
  items: CartItemResponse[];
  originalTotal: number;
  finalTotal: number;
  totalDiscount: number;
  storeId: string;
}

export interface CartItemResponse {
  itemId: UUID;
  productId: UUID;
  productName: string;
  productImageUrl: string;
  quantity: number;
  unitPrice: number;
  originalLineTotal: number;
  discountAmount: number;
  finalLineTotal: number;
  appliedOfferLabel: string;
}
